storage/postgres/admin_storage_test.go

```diff
@@ -84,6 +85,119 @@ func TestCreateTreeInvalidStates(t *testing.T) {
 	}
 }
 
+func TestAdminTX_TreeWithNulls(t *testing.T) {
+	cleanTestDB(DB, t)
+	s := postgres.NewAdminStorage(DB)
+	ctx := context.Background()
+
+	// Setup: create a tree and set all nullable columns to null.
+	// Some columns have to be manually updated, as it's not possible to set
+	// some proto fields to nil.
+	tree, err := storage.CreateTree(ctx, s, testonly.LogTree)
+	if err != nil {
+		t.Fatalf("CreateTree() failed: %v", err)
+	}
+	treeID := tree.TreeId
+
+	if err := setNulls(ctx, DB, treeID); err != nil {
+		t.Fatalf("setNulls() = %v, want = nil", err)
+	}
+
+	tests := []struct {
+		desc string
+		fn   storage.AdminTXFunc
+	}{
+		{
+			desc: "GetTree",
+			fn: func(ctx context.Context, tx storage.AdminTX) error {
+				_, err := tx.GetTree(ctx, treeID)
+				return err
+			},
+		},
+		{
+			// ListTreeIDs *shouldn't* care about other columns, but let's test it just
+			// in case.
+			desc: "ListTreeIDs",
+			fn: func(ctx context.Context, tx storage.AdminTX) error {
+				ids, err := tx.ListTreeIDs(ctx, false /* includeDeleted */)
+				if err != nil {
+					return err
+				}
+				for _, id := range ids {
+					if id == treeID {
+						return nil
+					}
+				}
+				return fmt.Errorf("ID not found: %v", treeID)
+			},
+		},
+		{
+			desc: "ListTrees",
+			fn: func(ctx context.Context, tx storage.AdminTX) error {
+				trees, err := tx.ListTrees(ctx, false /* includeDeleted */)
+				if err != nil {
+					return err
+				}
+				for _, tree := range trees {
+					if tree.TreeId == treeID {
+						return nil
+					}
+				}
+				return fmt.Errorf("ID not found: %v", treeID)
+			},
+		},
+	}
+	for _, test := range tests {
+		if err := s.ReadWriteTransaction(ctx, test.fn); err != nil {
+			t.Errorf("%v: err = %v, want = nil", test.desc, err)
+		}
+	}
+}
+
+func TestAdminTX_StorageSettingsNotSupported(t *testing.T) {
+	cleanTestDB(DB, t)
+	s := postgres.NewAdminStorage(DB)
+	ctx := context.Background()
+
+	settings, err := ptypes.MarshalAny(&keyspb.PEMKeyFile{})
```

Comment:
Are you just using `keyspb.PEMKeyFile` here as a convenient empty proto? If so, it'd be a bit clearer to use the [empty.Empty](https://godoc.org/github.com/golang/protobuf/ptypes/empty#Empty) proto.